### PR TITLE
TT-1620 Adding database layer to the local matching service and matching based on PID value

### PR DESCRIPTION
**Pull request by SKeerthana**
- Configured the service to use SQLite database so that anyone who wants to run this repo locally does not have to install any database.
- Endpoint returns match response only when Pid exists in the table, otherwise its no-match.
- Added integration test using h2 in-memory DB

### CODE REVIEW

File: src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/dao/VerifiedPidDAO.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.ida.verifylocalmatchingserviceexample.dao;
+
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+
+public interface VerifiedPidDAO {
```
Comment (RachaelBooth) [inline src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/dao/VerifiedPidDAO.java]: Can we have an easier to read name for this? Maybe VerifiedPid would do?

File: src/integration-test/java/uk/gov/ida/verifylocalmatchingserviceexample/rules/H2JDBIRule.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package uk.gov.ida.verifylocalmatchingserviceexample.rules;
+
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.rules.ExternalResource;
+import uk.gov.ida.verifylocalmatchingserviceexample.configuration.VerifyLocalMatchingServiceExampleConfiguration;
+
+public class H2JDBIRule extends ExternalResource {
```
Comment (RachaelBooth) [inline src/integration-test/java/uk/gov/ida/verifylocalmatchingserviceexample/rules/H2JDBIRule.java]: Can we have a more explanatory name for this? (I'm not sure what H2 means here?)

Comment (SKeerthana) [reply src/integration-test/java/uk/gov/ida/verifylocalmatchingserviceexample/rules/H2JDBIRule.java]: H2 is the in-memory database that we use for integration test. I will change it to TestDatabaseRule. Does that sound meaningful ?

File: src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/dao/VerifiedPidDAO.java (hunk 1 above)
Comment (SKeerthana) [reply src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/dao/VerifiedPidDAO.java]: Fair point. Will change it.

**Pull request merged**